ooniprobe/Model/NetworkMeasurement.mm

```diff
@@ -604,37 +433,39 @@ -(void)run{
 }
 
 -(void) run_test {
-    [super run_test];
-    setup_idempotent();
-    mk::nettests::DashTest()
-    .set_options("geoip_country_path", [geoip_country UTF8String])
-    .set_options("geoip_asn_path", [geoip_asn UTF8String])
-    .set_options("save_real_probe_ip", include_ip)
-    .set_options("save_real_probe_asn", include_asn)
-    .set_options("save_real_probe_cc", include_cc)
-    .set_options("no_collector", !upload_results)
-    .set_options("software_name", [@"ooniprobe-ios" UTF8String])
-    .set_options("software_version", [software_version UTF8String])
-    .set_output_filepath([[self getFileName:@"json"] UTF8String])
-    .set_error_filepath([[self getFileName:@"log"] UTF8String])
-    .set_verbosity(VERBOSITY)
-    .on_progress([self](double prog, const char *s) {
-        [self updateProgress:prog];
-    })
-    .on_log([self](uint32_t type, const char *s) {
-#ifdef DEBUG
-        NSLog(@"%s", s);
-#endif
-    })
-    .on_entry([self](std::string s) {
-        // Note: here we're reusing NDT function since the entry has
-        // basically the same characteristics, as far as deciding the
-        // color of the test line is concerned.
-        [self on_entry_ndt:s.c_str()];
-    })
-    .start([self]() {
+    mk::nettests::DashTest test;
+    [super run_test:test];
+    test.on_entry([self](std::string s) {
+        [self on_entry:s.c_str()];
+    });
+    test.start([self]() {
         [self testEnded];
     });
 }
 
+/*
+ Note: here we're reusing NDT function since the entry has
+ basically the same characteristics, as far as deciding the
+ color of the test line is concerned.
+ if the "failure" key exists and is not null then anomaly will be set to 1 (orange)
+ */
```

Comment:
If it's really equal, wouldn't it be possible to really invoke the same method? (I may be missing come ctx here)

Reply:
Only if we move it in the super class. I want to keep class specific methods. Nevermind the algorithm is the same.